kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -886,81 +909,94 @@ class SparseMatrixMultiplicationUtility
         * Merging by pairs allows to work with short rows as often as possible.
         */
         // Merge first two.
-        Idx a1 = *acol++;
-        Idx a2 = *acol++;
-        Idx c_col1 = MergeRows<true>(  bcol + bptr[a1], bcol + bptr[a1+1], bcol + bptr[a2], bcol + bptr[a2+1], tmp_col1 ) - tmp_col1;
+        TIndex a1 = *AColumn++;
+        TIndex a2 = *AColumn++;
+        TIndex c_col1 = MergeRows<true>(  BColumn + BPtr[a1], BColumn + BPtr[a1+1], BColumn + BPtr[a2], BColumn + BPtr[a2+1], Tmp1Column ) - Tmp1Column;
 
         // Go by pairs.
-        while(acol + 1 < acol_end) {
-            a1 = *acol++;
-            a2 = *acol++;
+        while(AColumn + 1 < AColumnEnd) {
+            a1 = *AColumn++;
+            a2 = *AColumn++;
 
-            Idx c_col2 = MergeRows<true>(  bcol + bptr[a1], bcol + bptr[a1+1], bcol + bptr[a2], bcol + bptr[a2+1], tmp_col2 ) - tmp_col2;
+            TIndex c_col2 = MergeRows<true>(  BColumn + BPtr[a1], BColumn + BPtr[a1+1], BColumn + BPtr[a2], BColumn + BPtr[a2+1], Tmp2Column ) - Tmp2Column;
 
-            if (acol == acol_end) {
-                return MergeRows<false>( tmp_col1, tmp_col1 + c_col1, tmp_col2, tmp_col2 + c_col2, tmp_col3 ) - tmp_col3;
+            if (AColumn == AColumnEnd) {
+                return MergeRows<false>( Tmp1Column, Tmp1Column + c_col1, Tmp2Column, Tmp2Column + c_col2, Tmp3Column ) - Tmp3Column;
             } else {
-                c_col1 = MergeRows<true>( tmp_col1, tmp_col1 + c_col1, tmp_col2, tmp_col2 + c_col2, tmp_col3 ) - tmp_col3;
+                c_col1 = MergeRows<true>( Tmp1Column, Tmp1Column + c_col1, Tmp2Column, Tmp2Column + c_col2, Tmp3Column ) - Tmp3Column;
 
-                std::swap(tmp_col1, tmp_col3);
+                std::swap(Tmp1Column, Tmp3Column);
             }
         }
 
         // Merge the tail.
-        a2 = *acol;
-        return MergeRows<false>( tmp_col1, tmp_col1 + c_col1, bcol + bptr[a2], bcol + bptr[a2+1], tmp_col2 ) - tmp_col2;
+        a2 = *AColumn;
```

Comment:
as usually...by changing the name to the variables you changed something that was relatively easy to review into something that is very hard to follow.

PLEASE avoid doing it!!!

Reply:
Sorry, you are right